### ISSUE TITLE
[AVM Module Issue]: 🐛 role assignments for NIC's does not work

### ISSUE THREAD

**Issue by Laffs2k5** [Azure/terraform-azurerm-avm-res-compute-virtualmachine#68]
### Check for previous/existing GitHub issues

- [X] I have checked for previous/existing GitHub issues

### Issue Type?

Bug

### (Optional) Module Version

0.12.0

### Description

Specifying role assignments for network interfaces does not work.

Attempting so causes errors like these:

```log
 Error: Unsupported attribute
│ 
│   on .terraform/modules/mgmt_windows_vm/main.networking.tf line 93, in resource "azurerm_role_assignment" "this_network_interface":
│   93:   role_definition_name                   = (length(split("/", each.value.role_definition_id_or_name))) > 3 ? null : each.value.role_assignment.role_definition_id_or_name
│     ├────────────────
│     │ each.value is object with 3 attributes
│ 
│ This object does not have an attribute named "role_definition_id_or_name".
```

Example of `network_interfaces`  part of vm declaration:

```hcl
  network_interfaces = {
    nic1 = {
      name = "nic-my"
      ip_configurations = {
        ip_configuration_1 = {
          name                          = "vm-my-ipconfig1"
          private_ip_subnet_resource_id = var.subnet_id
        }
      }
      role_assignments = { for user_key, principal_id in var.all_vm_users :
        "${user_key}_read_nic_of_vm-my" => {
          principal_id               = principal_id
          role_definition_id_or_name = "Reader"
        }
      }
    }
  }
```

The issue seems to be here: https://github.com/Azure/terraform-azurerm-avm-res-compute-virtualmachine/blob/862ab3a36ace3192f4fe55e3ad99f3914eaf71ee/main.networking.tf#L92-L93

```hcl
  role_definition_id                     = (length(split("/", each.value.role_definition_id_or_name))) > 3 ? each.value.role_assignment.role_definition_id_or_name : null
  role_definition_name                   = (length(split("/", each.value.role_definition_id_or_name))) > 3 ? null : each.value.role_assignment.role_definition_id_or_name
```

should be :

```hcl
  role_definition_id                     = (length(split("/", each.value.role_assignment.role_definition_id_or_name))) > 3 ? each.value.role_assignment.role_definition_id_or_name : null
  role_definition_name                   = (length(split("/", each.value.role_assignment.role_definition_id_or_name))) > 3 ? null : each.value.role_assignment.role_definition_id_or_name
```

Notice how `each.value.role_definition_id_or_name` changed to `each.value.role_assignment.role_definition_id_or_name`.

**Comment by jchancellor-ms** [comment]
@Laffs2k5 - Thanks for submitting this.  Let me get a fix in, and I'll add additional role assignments to the RBAC example so we get good testing of this going forward.  I'll look to push the fix in the next release later this week.

**Comment by jchancellor-ms** [comment]
@Laffs2k5 - This issue was corrected in release v0.13.0.  Please re-open this if you determine anything is still missing.
